src/container/Login.container.jsx

```diff
@@ -28,25 +35,7 @@ const LoginContainer = ({setToken}) => {
             return;
         }
 
-        const response = await getToken(form.username.value, form.password.value);
-
-        if (!response.success) {
-            if (response.status === 404 || response.status === 403 || response.status === 401) {
-                setForm({...form, error: "Invalid username/password!"});
-            } else {
-                setForm({...form, error: "Unknown server error!"});
-            }
-            return;
-        }
-
-        const token = response.data['token'];
-        if (token === undefined) {
-            setForm({...form, error: "Unknown server error!"});
-            return;
-        }
-
-        setToken(token);
-        setAuthTokenCookie(token)
+        dispatch(getAuthToken(form.username.value, form.password.value, navigate, handleOnError));
```

Comment:
Can you not rather pass through the entire form as a value/payload? i.e
```suggestion
        dispatch(getAuthToken(form, navigate, handleOnError));
```
If you need to send a different data structure to the API, you create a mapping function that maps the form values to the request values.